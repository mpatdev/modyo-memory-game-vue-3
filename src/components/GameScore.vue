<script setup>

</script>

<template>
  <div>
    <div>
      <h1>
        Bienvenido al juego de memoria
      </h1>
      <p>
        !Hola, Martin PatÂ¡
      </p>
    </div>

    <div class="flex gap-4">
      <div class="w-8/12">
        <div>
          <span>
            Intentos: 0
          </span>
          <span>
            Aciertos
          </span>
          <span>
            Tiempo: 00:00
          </span>
        </div>
      </div>

      <div class="w-4/12">
        Score: 10 Aciertos en 00:00
      </div>
    </div>
  </div>
</template>