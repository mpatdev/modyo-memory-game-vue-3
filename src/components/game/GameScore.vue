<script setup>
// imports
import useScoreStore from '@/store/score'
import { storeToRefs } from 'pinia';
import { computed } from 'vue';

import { formatTime } from '@/utils/time'

// Props
defineProps({
  player: {
    type: String,
    default: ''
  }
})

const { tries, hits, fails, time, history } = storeToRefs(useScoreStore())

const timeFormated = computed(() => formatTime(time.value))
</script>

<template>
  <div class="game-score">
    <div>
      <h1 class="font-medium text-lg uppercase">
        Hi, {{ player }}
      </h1>
    </div>

    <div class="flex gap-4">
      <div class="sm:w-8/12 w-full">
        <div>
          <p>
            Tries: {{ tries }} | Hits: {{ hits }} | Fails: {{ fails }} | Time: {{ timeFormated }}
          </p>
        </div>
      </div>

      <div
        v-if="history.tries !== 0"
        class="sm:w-4/12 flex sm:justify-end w-full"
      >
        Best score: Hits: {{ history.hits }} | Fails: {{ history.fails }} | Tries: {{ history.tries }} | {{ history.time }}
      </div>
    </div>
  </div>
</template>

<style lang="scss">
.game-score {
  @apply bg-secondary p-4 rounded border border-violet;
}
</style>