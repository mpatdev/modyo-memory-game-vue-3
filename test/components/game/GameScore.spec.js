import GameScore from '@/components/game/GameScore.vue'

import snapshotTest from '@test/snapshotTest'

snapshotTest(GameScore, 'GameScore')