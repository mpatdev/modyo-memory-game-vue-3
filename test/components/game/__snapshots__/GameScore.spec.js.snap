// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameScore > Check Snapshot 1`] = `
"<div class=\\"game-score\\">
  <div>
    <h1 class=\\"font-medium text-lg uppercase\\"> Hi, </h1>
  </div>
  <div class=\\"flex gap-4\\">
    <div class=\\"sm:w-8/12 w-full\\">
      <div>
        <p> Tries: 0 | Hits: 0 | Fails: 0 | Time: 00:00</p>
      </div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;
